src/pymediainfo/__init__.py

```diff
@@ -350,22 +349,61 @@ def can_parse(cls, library_file: Optional[str] = None) -> bool:
         except Exception:  # pylint: disable=broad-except
             return False
 
+    @overload
+    @classmethod
+    def parse(
+        # pylint: disable=too-many-arguments, too-many-locals
+        # pylint: disable=too-many-branches, too-many-statements
+        cls,
+        filename: Any,
+        *,
+        library_file: str | None = None,
+        cover_data: bool = False,
+        encoding_errors: str = "strict",
+        parse_speed: float = 0.5,
+        full: bool = True,
+        legacy_stream_display: bool = False,
+        mediainfo_options: dict[str, str] | None = None,
+        output: str,
+        buffer_size: int | None = 64 * 1024,
+    ) -> str: ...
+
+    @overload
+    @classmethod
+    def parse(
+        # pylint: disable=too-many-arguments, too-many-locals
+        # pylint: disable=too-many-branches, too-many-statements
+        cls,
+        filename: Any,
+        *,
+        library_file: str | None = None,
+        cover_data: bool = False,
+        encoding_errors: str = "strict",
+        parse_speed: float = 0.5,
+        full: bool = True,
+        legacy_stream_display: bool = False,
+        mediainfo_options: dict[str, str] | None = None,
+        output: None = None,
+        buffer_size: int | None = 64 * 1024,
+    ) -> MediaInfo: ...
+
     @classmethod
     def parse(
-        # pylint: disable=too-many-statements
-        # pylint: disable=too-many-branches, too-many-locals, too-many-arguments
+        # pylint: disable=too-many-arguments, too-many-locals
+        # pylint: disable=too-many-branches, too-many-statements
         cls,
         filename: Any,
-        library_file: Optional[str] = None,
+        *,
```

Comment:
Won't we break existing code by requesting other parameters to be keyword only (I mean it's not a big deal, we'll create a new release)? Was this required by a linter? If so, could you add a comment?

Reply:
Ruff has a rule for it: https://docs.astral.sh/ruff/rules/#flake8-boolean-trap-fbt.